sdk/resourcemanager/Azure.ResourceManager.Core/docs/MigratingFromCurrentToPreview.md

```diff
@@ -0,0 +1,86 @@
+# Differences between current and preview .NET Management SDK
+
+## Structured Resource Identifier
+In the past, the resource identifier returned by Azure (often in a property named `Id`) was returned as a string. This made it difficult for users to retrieve specific pieces of the identifier, since they needed to implement their own parsing logic. Today, you can instead implicitly cast the string into an object, which will do the parsing for you. There are three types of ResourceIdentifiers, and they correspond to the level at which the resource lives. A resource that lives on a tenant will have a **TenantResourceIdentifier**. A resource that lives under a subscription will have a **SubscriptionResourceIdentifer**. A resource that lives under a resource group will have a **ResourceGroupResourceIdentifier**.
+
+You can usually tell by the `id` string itself which type it is. If you're unsure, you can cast it onto a **ResourceIdentifier** and use the `Try` methods to retrieve the values.
+
+### Casting to a specific type
+```csharp
+string resourceId = "/subscriptions/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/resourceGroups/workshop2021-rg/providers/Microsoft.Network/virtualNetworks/myVnet/subnets/mySubnet";
+// We know the subnet is a resource group level identifier since it has a resource group name in its string
+ResourceGroupResourceIdentifier id = resourceId;
+Console.WriteLine($"Subscription: {id.SubscriptionId}");
+Console.WriteLine($"ResourceGroup: {id.ResourceGroupName}");
+Console.WriteLine($"Vnet: {id.Parent.Name}");
+Console.WriteLine($"Subnet: {id.Name}");
+```
+
+### Casting to the base resource identifier
+```csharp
+string resourceId = "/subscriptions/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/resourceGroups/workshop2021-rg/providers/Microsoft.Network/virtualNetworks/myVnet/subnets/mySubnet";
+// Assume we don't know what type of resource id we have we can cast to the base type
+ResourceIdentifier id = resourceId;
+string property;
+if (id.TryGetSubscriptionId(out property))
+    Console.WriteLine($"Subscription: {property}");
+if (id.TryGetResourceGroupName(out property))
+    Console.WriteLine($"ResourceGroup: {property}");
+Console.WriteLine($"Vnet: {id.Parent.Name}");
+Console.WriteLine($"Subnet: {id.Name}");
+```
+
+## [Resource]Operations
+
+This represents a service client that is scoped to a particular resource.
+You can directly execute all operations on that client without needing to pass in scope
+parameters such as subscription id or resource name.
+
+*Old*
+```csharp
+    ComputeManagementClient computeClient = new ComputeManagementClient(subscriptionId, new DefaultAzureCredential());
```

Comment:
these examples are not parallel.  Where did the subscriptionId come from in this example?  Why is it not used in the example below?